### PR TITLE
Upgrade libcusparse to 13-2 to support CUDA 13 base image

### LINKED ISSUE #3823: MaxText Docker Image Build Failure
GitHub Actions workflow [Build Images #1002](https://github.com/AI-Hypercomputer/maxtext/actions/runs/25409857331) failed.

Event: schedule
Branch: [main](https://github.com/AI-Hypercomputer/maxtext/tree/main)
Commit: [211e5389943f0a98c8658009a1b11b6ae9d1b12c](https://github.com/AI-Hypercomputer/maxtext/commit/211e5389943f0a98c8658009a1b11b6ae9d1b12c)

<sup><i>Created by [jayqi/failed-build-issue-action](https://github.com/jayqi/failed-build-issue-action)</i></sup>

### PR DESCRIPTION
**Pull request by darisoy**
# Description

This PR upgrades the `libcusparse` package version installed in the GPU dependencies Dockerfile to support the recently updated CUDA 13 base image, resolving the daily Docker build failures.
**Why is this change being made?**
The daily GitHub Actions workflow **Build Images (#1002)** started failing for GPU jobs (`maxtext_gpu_jax_stable` and `maxtext_gpu_jax_nightly`) because the package `libcusparse-12-6` could not be located. 
**Problem solved and context:**
The base image `ghcr.io/nvidia/jax:base` was recently upgraded to use **Ubuntu 24.04 (Noble)** and **CUDA 13.2.1** (upgraded from CUDA 12). Consequently, CUDA 12 packages (like `libcusparse-12-6`) are no longer available in the repositories. 

**Specific Implementation:**
- **Issue 1 Fix:** Diagnostic package search (`apt-cache search libcusparse`) inside the `ghcr.io/nvidia/jax:base` image confirmed that the correct runtime package name for CUDA 13.2 is **`libcusparse-13-2`**. We updated `src/dependencies/dockerfiles/maxtext_gpu_dependencies.Dockerfile` to install `libcusparse-13-2` instead of `libcusparse-12-6`.
- **Issue 2 Fix:** Eagerly imported `torch` in the top-level imports section of `to_maxtext.py` instead of lazy loading it, resolving the import resolution conflict.

FIXES: b/510149167
FIXES: #3823

# Tests

The changes were verified as follows:
1. **Base Image Inspection:** Verified inside a running instance of the base image `ghcr.io/nvidia/jax:base` that `libcusparse-13-2` is the correct and installable package name:
   ```bash
   docker run --rm ghcr.io/nvidia/jax:base apt-cache search libcusparse
   ```

Output:

```
libcusparse-13-2 - CUSPARSE native runtime libraries
libcusparse-dev-13-2 - CUSPARSE native dev links, headers
```

2. Workflow Path Validation: Confirmed `.github/workflows/UploadDockerImages.yml` correctly builds the Dockerfile from the package wheel, which will automatically propagate this fix during the next scheduled workflow.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run end-to-end tests tests and provided workload links above if applicable.
- [X] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/maxtext/checkpoint_conversion/to_maxtext.py [hunk 1]
```diff
@@ -77,6 +77,7 @@
 import numpy as np
 from orbax.checkpoint import type_handlers
 from safetensors import safe_open
+import torch
```
Comment (SurbhiJainUSC) [inline src/maxtext/checkpoint_conversion/to_maxtext.py]: We need to do lazy import for CI tests.

File: src/dependencies/dockerfiles/maxtext_gpu_dependencies.Dockerfile [hunk 2]
```diff
@@ -27,7 +27,7 @@ RUN apt-get update && apt-get install -y google-cloud-sdk
 ENV PATH="/usr/local/google-cloud-sdk/bin:${PATH}"
 
 # Upgrade libcusprase to work with Jax
-RUN apt-get update && apt-get install -y libcusparse-12-6
+RUN apt-get update && apt-get install -y libcusparse-13-2
```
Comment (bvandermoon) [inline src/dependencies/dockerfiles/maxtext_gpu_dependencies.Dockerfile:30]: @SurbhiJainUSC is CI passing enough to test this change? Or do we need to run a real workflow also?

**Review by hengtaoguo — APPROVED** [review]
Thanks for the fix!

File: src/dependencies/dockerfiles/maxtext_gpu_dependencies.Dockerfile (hunk 2 above)
Comment (SurbhiJainUSC) [reply src/dependencies/dockerfiles/maxtext_gpu_dependencies.Dockerfile:30]: @darisoy - please run [this](https://github.com/AI-Hypercomputer/maxtext/actions/workflows/UploadDockerImages.yml) workflow for GPU to verify the fix.

**Review by SurbhiJainUSC — APPROVED** [review]

File: src/dependencies/dockerfiles/maxtext_gpu_dependencies.Dockerfile (hunk 2 above)
Comment (darisoy) [reply src/dependencies/dockerfiles/maxtext_gpu_dependencies.Dockerfile:30]: gpu image is building fine in https://github.com/AI-Hypercomputer/maxtext/actions/runs/26063728802/job/76630499419 (other failures in this workflow are unrelated/expected)
